### DISCUSSION TITLE
Error with Async Dynamic APIs

### DISCUSSION THREAD

**Discussion by todor0v** [payloadcms/payload#8713] [Q&A]
### Environment Info

Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: N/A
  pnpm: 9.12.1
Relevant Packages:
  payload: 3.0.0-beta.111
  next: 15.0.0-canary.173
  @payloadcms/db-mongodb: 3.0.0-beta.111
  @payloadcms/graphql: 3.0.0-beta.111
  @payloadcms/next/utilities: 3.0.0-beta.111
  @payloadcms/plugin-cloud-storage: 3.0.0-beta.111
  @payloadcms/richtext-lexical: 3.0.0-beta.111
  @payloadcms/storage-s3: 3.0.0-beta.111
  @payloadcms/translations: 3.0.0-beta.111
  @payloadcms/ui/shared: 3.0.0-beta.111
  react: 19.0.0-rc-3edc000d-20240926
  react-dom: 19.0.0-rc-3edc000d-20240926

### Describe the Bug

When opening the Payload admin panel using the versions described in the environment info, I get the following errors:
![Image](https://github.com/user-attachments/assets/9d4b0d59-95cf-490c-adae-1d9cda464a9f)

### Reproduction Steps

I am aware that we closed https://github.com/payloadcms/payload/issues/8463, however, I still get the above mentioned errors. Solutions I've tried:
- I deleted `node_modules` and `.next` and installed again.
- I made sure the content of all payload files is absolutely exactly the same as [these](https://github.com/payloadcms/payload/tree/beta/templates/website/src/app/(payload)).
- I do not have any usage of `next/headers`

### Adapters and Plugins

db-mongodb

**Comment by jmikrut** [comment]
@todor0v is it possible that this is coming from some custom code in your project? We don't have a single reference to `x-forwarded-proto` at all in our code.

A new blank Payload project with your versions above does not show this behavior. We will need a reproduction in order to help here. Will convert to Q&A discussion as I don't believe this is an issue with Payload, but if it is, we will re-open / fix accordingly.
